.github/workflows/publish.yml

```diff
@@ -0,0 +1,39 @@
+name: publish
+
+on:
+  push:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "src/**"
+      - ".github/workflows/publish.yml"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        machine: [windows-latest, macos-latest, ubuntu-20.04]
+        architecture: [x64, arm64]
+        include:
+          - machine: windows-latest
+            platform: win
+          - machine: macos-latest
+            platform: osx
+          - machine: ubuntu-20.04
+            platform: linux
+    runs-on: ${{ matrix.machine }}
+    timeout-minutes: 5
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1
+      - name: install linux prerequisites
+        if: matrix.platform == "linux"
+        run: |
+          sudo apt-get install clang zlib1g-dev
+          sudo apk add clang build-base zlib-dev
+      - name: publish
+        run: dotnet publish -r ${{ matrix.platform }}-${{ matrix.architecture }} -c Release
```

Comment:
https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/compiling.md#cross-architecture-compilation


The bottom of that section has this and I'm not sure how to find that directory on a github worker.
> You also need to specify the sysroot directory for Clang using the SysRoot property. For example, assuming you are using one of ARM64-targeting [Docker images](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/coreclr/linux-instructions.md#Docker-Images) employed for cross-compilation by this repo, you may publish for linux-arm64 with the following command:
>dotnet publish -r linux-arm64 -c Release -p:CppCompilerAndLinker=clang-9 -p:SysRoot=/crossrootfs/arm64

Reply:
Should we be using those docker files somehow

Reply:
if it's too much hassle we can forego cross-arch compilation for now.
we don't need arm builds for feature parity with the Go version.

Reply:
Yea I'm a little confused while reading the docs. Maybe I can look into it again later on